tests/pyfunc/test_chat_model.py

```diff
@@ -87,6 +96,28 @@ def test_chat_model_save_load(tmp_path):
     assert output_schema == CHAT_MODEL_OUTPUT_SCHEMA
 
 
+def test_chat_model_with_trace(tmp_path):
+    model = ChatModelWithTrace()
+    mlflow.pyfunc.save_model(python_model=model, path=tmp_path)
+
+    # predict() call during saving chat model should not generate a trace
+    assert len(get_traces()) == 0
+
+    loaded_model = mlflow.pyfunc.load_model(tmp_path)
+    messages = [
+        {"role": "system", "content": "You are a helpful assistant"},
+        {"role": "user", "content": "Hello!"},
+    ]
+    loaded_model.predict({"messages": messages})
+
+    traces = get_traces()
+    assert len(traces) == 1
+    assert traces[0].info.tags[TraceTagKey.TRACE_NAME] == "predict"
+    request = json.loads(traces[0].data.request)
+    assert request["messages"] == [str(ChatMessage(**msg)) for msg in messages]
```

Comment:
unrelated: `str()` is required because dataclass e.g. `ChatMessage` is not JSON serializable so MLflow falls back to string serialization when saving them. We may want to make it serialized as dictionary not str to make deserialization easier.

Reply:
ChatMessage has `to_dict` method maybe we can reuse?

Reply:
On second thought, we may not want to touch the serialization... Showing the fact that the model receives `ChatMessage` object instead of message dictionaries that was originally passed as request, is an important information for debugging.